src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -75,11 +79,49 @@ void handle_reshape::run(program& p) {
             // vector for storing nodes that are reorder type, for which splitted primitives are needed (except for the
             // first one where orginal reshape will be used)
             std::vector<program_node*> reorder_node_to_split;
+            std::vector<program_node*> onednn_users;
 
             // find the users of reshape that are reorder type, if none present then skip the current node
+            // find users who are onednn impl
             for (const auto& user : node->get_users()) {
                 if (user->is_type<reorder>())
                     reorder_node_to_split.push_back(user);
+                if (user->get_preferred_impl_type() == cldnn::impl_types::onednn)
+                    onednn_users.push_back(user);
+            }
+
+            // If onednn user doesn't support new input data type from future "reorder:_reshape_input_" reorder,
+            // remove target reorder_node to keep original datatype
+            if (!onednn_users.empty() && !reorder_node_to_split.empty()) {
+                // Copy reorder_node_to_split to iteration
+                std::vector<program_node*> reorder_users(reorder_node_to_split);
+                for (const auto& reorder_node : reorder_users) {
+                    auto output_data_type = reorder_node->get_output_layout().data_type;
+                    bool onednn_support = true;
+                    for (const auto& user : onednn_users) {
+                        if (user->get_preferred_impl_type() == cldnn::impl_types::onednn) {
```

Comment:
(no need to apply in this PR) i think this line can be removed.

Reply:
Removed. Thanks!